### ISSUE TITLE
Include MacOS ARM64 wheel building in CI

### ISSUE THREAD

**Issue by gipert** [g4edge/pyg4ometry#74]
CI builds on ARM64 fail somehow with the error message first reported in https://github.com/g4edge/pyg4ometry/issues/49. Needs investigation.

**Comment by stewartboogert** [comment]
macOS arm64 wheels have been uploaded to pypi for pyg4ometry v1.1.0 and  python 3.8, 3.9, 3.10 and 3.11

**Comment by gipert** [comment]
@stewartboogert could you post the error message here?

**Comment by stewartboogert** [comment]
Still problematic with macOS arm64 (so built using pypa/cibuildwheel@v2.16.2). Error seen on import 

```
ImportError: dlopen(/Users/stewart.boogert/venv/python3.11-pyg4-pypi/lib/python3.11/site-packages/pyg4ometry/pycgal/CGAL.cpython-311-darwin.so, 0x0002): symbol not found in flat namespace '___gmpn_add_n'
```

Should create a .dylib directory (confirmed on macOS intel) which contains the dependencies. This is not present on arm64 build. otool does not show the libraries required 

```
(python3.11-pyg4-pypi) DLAST0155% otool -L /Users/stewart.boogert/venv/python3.11-pyg4-pypi/lib/python3.11/site-packages/pyg4ometry/pycgal/CGAL.cpython-311-darwin.so
/Users/stewart.boogert/venv/python3.11-pyg4-pypi/lib/python3.11/site-packages/pyg4ometry/pycgal/CGAL.cpython-311-darwin.so:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.36.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
```

**Comment by gipert** [comment]
I would open an issue on the cbuildwheel repository.

**Comment by stewartboogert** [comment]
Trying a few things first, but I think I will.

**Comment by gipert** [comment]
New wheels built with the latest cbuildwheel appearing soon on PyPI... let's see if they are still broken

**Comment by stewartboogert** [comment]
Ok there are still problems with the .dylib dependency directory. So running cibuildwheel locally on my apple silicon M1 I find the .dylib dependency directory and don't find it in the intel directory. So the other way around from what we observe on the GitHub runners. 

So now I can properly build the wheels with all the appropriate deps. My previous apple silicon builds did not have the deps and would often fail as users would need the correct dependencies. So makes sense but strange behaviour from cibuildwheel as this would often come up I suspect

**Comment by gipert** [comment]
We've decided to upload ARM64 wheels manually for now, but let's keep this open.

**Comment by stewartboogert** [comment]
Manually uploaded, let’s see if users have issues

**Comment by stewartboogert** [comment]
Fixed in #177 by matrix of os in GitHub Actions and not using cbuildwheel to creation of the wheels
